File: Diagrammatize.Design.Pattern/code/cxx/AbstractFactory/Sample/listfactory/ListPage.cpp
#include "ListPage.hpp"
#include <sstream>

ListPage::ListPage(const std::string& title, const std::string& author): Page(title, author) {
}

std::string ListPage::makeHTML() {
    std::stringstream buffer;
    buffer << "<html><head><title>" << title << "</title></head>\n";
    buffer << "<body>\n";
    buffer << "<h1>" << title << "</h1>\n";
    buffer << "<ul>\n";
    for (auto item : content) {
        buffer << item->makeHTML();
    }
    buffer << "</ul>\n";
    buffer << "<hr><address>" << author << "</address>";
    buffer << "</body></html>\n";
    return buffer.str();
}

File: Diagrammatize.Design.Pattern/code/python/AbstractFactory/Sample/listfactory/ListPage.py
import io
from factory import Page

class ListPage(Page):
    def __init__(self, title, author):
        super().__init__(title, author)

    def makeHTML(self):
        buffer = io.StringIO()
        buffer.write("<html><head><title>" + self.title + "</title></head>\n")
        buffer.write("<body>\n")
        buffer.write("<h1>" + self.title + "</h1>\n")
        buffer.write("<ul>\n")
        for item in self.content:
            buffer.write(item.makeHTML())
        buffer.write("</ul>\n")
        buffer.write("<hr><address>" + self.author + "</address>")
        buffer.write("</body></html>\n")
        return buffer.getvalue()
